bin/tests/test_main.cpp

```diff
@@ -1,198 +0,0 @@
-/*************************************************************************/
-/* Copyright (c) 2007-2016 Juan Linietsky, Ariel Manzur.                 */
-/*                                                                       */
-/* Permission is hereby granted, free of charge, to any person obtaining */
-/* a copy of this software and associated documentation files (the       */
-/* "Software"), to deal in the Software without restriction, including   */
-/* without limitation the rights to use, copy, modify, merge, publish,   */
-/* distribute, sublicense, and/or sell copies of the Software, and to    */
-/* permit persons to whom the Software is furnished to do so, subject to */
-/* the following conditions:                                             */
-/*                                                                       */
-/* The above copyright notice and this permission notice shall be        */
-/* included in all copies or substantial portions of the Software.       */
-/*                                                                       */
-/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
-/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
-/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
-/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
-/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
-/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
-/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
-/*************************************************************************/
-#include "list.h"
-#include "os/main_loop.h"
-
-#ifdef DEBUG_ENABLED
-
-#include "test_string.h"
-#include "test_containers.h"
-#include "test_math.h"
-#include "test_gui.h"
-#include "test_render.h"
-#include "test_sound.h"
-#include "test_misc.h"
-#include "test_physics.h"
-#include "test_physics_2d.h"
-#include "test_python.h"
-#include "test_io.h"
-#include "test_particles.h"
-#include "test_detailer.h"
-#include "test_shader_lang.h"
-#include "test_gdscript.h"
-#include "test_image.h"
-
-
-const char ** tests_get_names()  {
-
-	static const char* test_names[]={
-		"string",
-		"containers",
-		"math",
-		"render",
-		"particles",
-		"multimesh",
-		"gui",
-		"io",
-		"shaderlang",
-		"physics",
-		NULL
-	};
-
-	return test_names;
-}
-
-MainLoop* test_main(String p_test,const List<String>& p_args) {
-
-
-	if (p_test=="string") {
-
-		return TestString::test();
-	}
-
-	if (p_test=="containers") {
-
-		return TestContainers::test();
-	}
-
-	if (p_test=="math") {
-
-		return TestMath::test();
-	}
-
-	if (p_test=="physics") {
-
-		return TestPhysics::test();
-	}
-
-	if (p_test=="physics_2d") {
-
-		return TestPhysics2D::test();
-	}
-
-  	if (p_test=="misc") {
-
-		return TestMisc::test();
-	}
-
-	if (p_test=="render") {
-
```

Comment:
(Since Noshyaar keeps closing this, I am going to continue to post it until one thread remains visible for discussion, no need to censor)

![](https://i.gyazo.com/364f1b11bc57924374d0a8c81df1eeab.png)

Lead developer sent this message after a user was giving his honest feedback via this [reddit thread](https://www.reddit.com/r/gamedev/comments/7gmtxl/godot_30_beta_1_is_there_right_in_time_for_ludum/dqk9nel/).

They also said (I kid you not) -- giving feedback is "**bullying**",  a "**professional whiner**", and "**ranting**". Reduz's passive aggressive tone and insulting behavior is rampant there.  I already terminated my patreon account because of this, and not saying you should either (FLOSS projects) are great, however, for the lead developer to act like this towards people giving feedback.. is not okay!

Let me re-iterate, not saying that you should stop giving feedback, but I just want this message to be seen across github and reach as far as possible. Because this kind of communication and how the lead developer treats a regular Godot developer is not right at all. I've never seen this behavior from a FLOSS project lead developer.  It's a shame, because the guy giving feedback is just a random person anyway, and the way the project managers and lead developers took it so personally like it was an attack (when it wasn't), means the lead developers/contributors are possibly not viewing your issues with the right mindset.  

Also, after reading the last paragraph from [this post by akien](https://www.reddit.com/r/gamedev/comments/7gmtxl/godot_30_beta_1_is_there_right_in_time_for_ludum/dqlqj2v/)
> From here on, we will avoid responding to xxx's threads on the various Godot communities, and focus instead on the huge amount of constructive feedback and bug reports that we get from other users. This will likely overlap with some of the issues mentioned above, which will thus be fixed eventually, but we won't accept being bullied into working on them now when we have different priorities. Users are free to disagree with our priorities, but in the end the contributors are the ones making the call.

He's saying they will now favor/prioritize issues based on the user reporting it, and not the issue itself. (Extreme prejudice), which will not be good for the project because the issue should be irrelevant to the user reporting it. Just because if a user who has a snotty attitude that is reporting a huge client crash error, doesn't mean that client crash error should not be taken lightly.



He's also stating that anyone giving hard feedback is now "bullying" them. Which is not true whatsoever.

Just wanting to post this to get the word out there, hopefully some things change. @reduz @akien-mga  This is just unbelievable :(